### ISSUE TITLE
Missing dll

### ISSUE THREAD

**Issue by diabl0w** [Samsung-Loki/Thor#1]
## Description
Linux.zip from Release 1.0.1

## Reproduction
1)
```
./ThorRewrite.Shell 
The application to execute does not exist: '/home/xxxxx/Samsung/ThorRewrite.Shell.dll'
```
2) the file is not present in the zip

## Expected behaviour
ThorRewrite.Shell.dll to be present

## Hardware and software
Linux

**Comment by TheAirBlow** [comment]
Looks like you're trying to run this .NET 7 executable with mono (as a .NET Framework application)? 
Make sure you have .NET 7 Runtime installed in your system. Closed as invalid.

**Comment by lmateovf** [comment]
Hi! 
First of all thank you for your work.
Even though I have .NET 7.0.306 installed (see picture) it outputs the same message as this person, i have rebooted and checked Microsoft's manual... nothing....
![Captura desde 2023-07-15 11-52-03](https://github.com/Samsung-Loki/Thor/assets/62722086/6e2ce2d0-e448-4af2-938d-4495a3e46ee7)

**Comment by lmateovf** [comment]
I must add that I got a little confused trying to install it,
- first I tried with your script install-dotnet.sh but it installed .NET 6.0
- second I tried with apt: missing package - wich I finally managed to install following Microsoft's manual
- I also tried installing manually, but I didn't like the idea of setting .NET on my PATH and HOME... so, as I said I finally installed it with apt...

**Comment by diabl0w** [comment]
i have NET7 framework installed

**Comment by ethical-haquer** [comment]
I can confirm this issue. The same thing happens when I run:
`./ThorRewrite.Shell`
I already have .NET installed and am able to run the "TheAirBlow.Thor.Shell" file just fine. @diabl0w, have you run that file? I'm not sure, but it almost looks like the "ThorRewrite.Shell" file is from a later version, that shouldn't have been included in the release.

**Comment by diabl0w** [comment]
> I can confirm this issue. The same thing happens when I run: `./ThorRewrite.Shell` I already have .NET installed and am able to run the "TheAirBlow.Thor.Shell" file just fine. @diabl0w, have you run that file? I'm not sure, but it almost looks like the "ThorRewrite.Shell" file is from an earlier version, that shouldn't have been included in the release.

you're correct, that binary does work and is probably what i shouldve run. i wasnt paying enough attention. non the less, this issue was marked as [invalid][wontfix] which may not be appropriate, probably needs those tags removed and reopened if not fixed already

**Comment by ethical-haquer** [comment]
#7

**Comment by diabl0w** [comment]
> #7

ah yes, i guess he took it more seriosuly for some reason when that guy reported it

**Comment by ethical-haquer** [comment]
Do you know how to mention an issue properly? :confused: It's my first time trying to, but when I use "#7" it doesn't look like the above issue mention. TIA.

**Comment by TheAirBlow** [comment]
If you get this error, run `TheAirBlow.Thor.Shell` instead. `ThorRewrite.Shell` is a leftover from an older version which I forgot to delete.

**Comment by ethical-haquer** [comment]
:+1: Perhaps how to run Thor itself should be added to the [Usage Guide](https://github.com/Samsung-Loki/Thor#usage-guide). Also, did you see my PM on XDA, #11, and [this post on XDA](https://forum.xda-developers.com/t/dev-thor-flash-utility-the-new-samsung-flash-tool.4597355/page-2#post-88756577)?

**Comment by TheAirBlow** [comment]
> 👍 Perhaps how to run Thor itself should be added to the [Usage Guide](https://github.com/Samsung-Loki/Thor#usage-guide). Also, did you see my PM on XDA, #11, and [this post on XDA](https://forum.xda-developers.com/t/dev-thor-flash-utility-the-new-samsung-flash-tool.4597355/page-2#post-88756577)?

Yes, Unknown2 being 80 is common and is something in the newer devices probably.

**Comment by ethical-haquer** [comment]
> > 👍 Perhaps how to run Thor itself should be added to the [Usage Guide](https://github.com/Samsung-Loki/Thor#usage-guide). Also, did you see my PM on XDA, #11, and [this post on XDA](https://forum.xda-developers.com/t/dev-thor-flash-utility-the-new-samsung-flash-tool.4597355/page-2#post-88756577)?
> 
> Yes, Unknown2 being 80 is common and is something in the newer devices probably.

I figured it was. :smile: I hope your other project goes well. See you!

**Comment by TheAirBlow** [comment]
> > > 👍 Perhaps how to run Thor itself should be added to the [Usage Guide](https://github.com/Samsung-Loki/Thor#usage-guide). Also, did you see my PM on XDA, #11, and [this post on XDA](https://forum.xda-developers.com/t/dev-thor-flash-utility-the-new-samsung-flash-tool.4597355/page-2#post-88756577)?
> > 
> > 
> > Yes, Unknown2 being 80 is common and is something in the newer devices probably.
> 
> I figured it was. 😄 I hope your other project goes well. See you!

can you send the bootloader image of your phone's firmware?

**Comment by ethical-haquer** [comment]
Certainly! I'm downloading my phone's firmware now (I've wanted to have it on hand anyways):
![Bifrost](https://github.com/Samsung-Loki/Thor/assets/141518185/29c4a55d-ff69-4b99-82b7-6c54629bceaf)

**Comment by ethical-haquer** [comment]
Here it is (minus NON-HLOS.bin.lz4, b/c that one file is 44Mib, and the maximum file size I'm allowed to upload is 25Mib. If you need that file though, just let me know :smile:):
[BL_A525FXXS6DWG2_A525FXXS6DWG2_MQB68570034_REV00_user_low_ship_MULTI_CERT.zip](https://github.com/Samsung-Loki/Thor/files/12481097/BL_A525FXXS6DWG2_A525FXXS6DWG2_MQB68570034_REV00_user_low_ship_MULTI_CERT.zip)

**Comment by al12gamer** [comment]
I'd like to add to this issue and note that dot net 7 has depreciated in 2025 and it seems we can no longer install it in Fedora 42. Does that make this tool useless?

**Comment by TheAirBlow** [comment]
I'll work on providing a .NET AOT build with .NET 9.

**Comment by TheAirBlow** [comment]
Check out release [1.1.0](https://github.com/Samsung-Loki/Thor/releases/tag/1.1.0)
